src/options.js

```diff
@@ -0,0 +1,159 @@
+  -p   --python     =/path/to/python2       -- python executable
+  -f   --flag       ="--expose-gc"          -- *v8 flags to include during compilation
+  -c   --configure  ="--with-dtrace"        -- *pass arguments to the configure command
+  -m   --make       ="--loglevel"           -- *pass arguments to the make command
+  -vc  --vcBuild    =x64                    -- *pass arguments to vcbuild.bat
+  -s   --snapshot   =/path/to/snapshot      -- build with warmup snapshot
+  -r   --resource   =./paths/**/*           -- *embed file bytes within the binary
+  -b   --bundle     =webpack-config.js      -- use default configuration or provide custom webpack config
+       --temp       =./path/to/temp         -- nexe temp files (for downloads and source builds)
+       --ico                                -- file name for alternate icon file (windows)
+       --rc-*                               -- populate rc file options (windows)
+       --clean                              -- force download of sources
+       --enableNodeCli                      -- enable node cli enforcement (blocks app cli)
+       --sourceUrl                          -- pass an alternate source (node.tar.gz) url
+       --silent                             -- disable logging
+       --verbose                            -- set logging to verbose
+
+       -* variable key name                 * option can be used more than once
+
+
+
+`.trim()
+
+function flattenFilter (...args) {
+  return [].concat(...args).filter(x => x)
+}
+
+/**
+ * Extract keys such as { "rc-CompanyName": "Node.js" } to
+ * { CompanyName: "Node.js" }
+ * @param {*} match
+ * @param {*} options
+ */
+function extractCliMap (match, options) {
+  return Object.keys(options).filter(x => match.test(x))
+    .reduce((map, option) => {
+      const key = option.split('-')[1]
+      map = map || {}
+      map[key] = options[option]
+      delete options[option]
+      return map
+    }, null) || {}
+}
+
+function tryResolveMainFileName () {
+  let filename = 'nexe'
+  try {
+    const file = require.resolve(process.cwd())
+    filename = basename(file).replace(extname(file), '')
+  } catch (_) {}
+
+  return filename === 'index' ? ('nexe_' + Date.now()) : filename
+}
+
+function extractLogLevel (options) {
+  if (options.loglevel) return options.loglevel
+  if (options.silent) return 'silent'
+  if (options.verbose) return 'verbose'
+  return 'info'
+}
+
+function extractName (options) {
+  let name = options.name
+  if (typeof options.input === 'string' && !name) {
+    name = basename(options.input).replace(extname(options.input), '')
+  }
+  name = name || tryResolveMainFileName()
+  return name.replace(/\.exe$/, '')
+}
+
+function normalizeOptionsAsync (input) {
+  if (argv.help || Boolean(argv._.filter(x => x === 'version')[0])) {
+    return Bluebird.fromCallback(cb => process.stderr.write(
+      argv.help ? help : '2.0.0-rc.1' + EOL,
```

Comment:
Yeah. That'd be own-package reflection. Also need local package reflection for `options.name`